test/Atmos/EDMF/closures/mixing_length.jl

```diff
@@ -0,0 +1,127 @@
+#### Mixing length model kernels
+
+"""
+    mixing_length(
+        m::AtmosModel{FT},
+        ml::MixingLengthModel,
+        state::Vars,
+        diffusive::Vars,
+        aux::Vars,
+        t::Real,
+        δ::Tuple,
+        εt::Tuple,
+        ts,
+    ) where {FT}
+
+Returns the mixing length used in the diffusive turbulence closure, given:
+ - `m`, an `AtmosModel`
+ - `ml`, a `MixingLengthModel`
+ - `state`, state variables
+ - `diffusive`, additional variables
+ - `aux`, auxiliary variables
+ - `t`, the time
+ - `δ`, the detrainment rate
+ - `εt`, the turbulent entrainment rate
+ - `ts`, NamedTuple of thermodynamic states
+"""
+function mixing_length(
+    m::AtmosModel{FT},
+    ml::MixingLengthModel,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    δ::Tuple,
+    εt::Tuple,
+    ts,
+) where {FT}
+
+    # need to code / use the functions: obukhov_length, ustar, ϕ_m
+    m.turbconv.surface
+    # Alias convention:
+    gm = state
+    en = state.turbconv.environment
+    up = state.turbconv.updraft
+    gm_d = diffusive
+    gm_a = aux
+    en_d = diffusive.turbconv.environment
+    N_up = n_updrafts(m.turbconv)
+
+    z = altitude(m, aux)
+    _grav::FT = grav(m.param_set)
+    ρinv = 1 / gm.ρ
+
+    # precompute
+    en_area = environment_area(state, aux, N_up)
+    w_env = environment_w(state, aux, N_up)
+
+    # TODO: check rank of `en_d.∇u`
+    Shear² = diffusive.turbconv.S²
+    tke = max(en.ρatke, 0) * ρinv / en_area
+
+    ustar = m.turbconv.surface.ustar
+    obukhov_length = m.turbconv.surface.obukhov_length
+
+    # buoyancy related functions
+    ∂b∂z, Nˢ_eff = compute_buoyancy_gradients(m, state, diffusive, aux, t, ts)
+    Grad_Ri = ∇Richardson_number(∂b∂z, Shear², 1 / ml.max_length, ml.Ri_c)
+    Pr_z = turbulent_Prandtl_number(ml.Pr_n, Grad_Ri, ml.ω_pr)
+
+    # compute L1
+    Nˢ_fact = (sign(Nˢ_eff - eps(FT)) + 1) / 2
+    coeff = min(sqrt(ml.c_b * tke) / Nˢ_eff, ml.max_length)
+    L_Nˢ = coeff * Nˢ_fact + ml.max_length * (FT(1) - Nˢ_fact)
+
+    # compute L2 - law of the wall
+    _, _, _, _, _, _, tke_surf = subdomain_surface_values(
+        m.turbconv.surface,
```

Comment:
Wouldn't be better to have this function return named tuple? So that you dont have to worry about the order of returned variables?